src/app/download-handler.ts

```diff
@@ -0,0 +1,168 @@
+     */
+    private static async showDialog(): Promise<void> {
+        const focusedWindow = BrowserWindow.getFocusedWindow();
+        const message = i18n.t('The file you are trying to open cannot be found in the specified path.', DOWNLOAD_MANAGER_NAMESPACE)();
+        const title = i18n.t('File not Found', DOWNLOAD_MANAGER_NAMESPACE)();
+
+        if (!focusedWindow || !windowExists(focusedWindow)) {
+            return;
+        }
+        await dialog.showMessageBox(focusedWindow, {
+            message,
+            title,
+            type: 'error',
+        });
+    }
+
+    private window!: Electron.WebContents | null;
+    private items: IDownloadManager[] = [];
+
+    /**
+     * Sets the window for the download handler
+     * @param window Window object
+     */
+    public setWindow(window: Electron.WebContents): void {
+        this.window = window;
+        logger.info(`download-handler: Initialized download handler`);
+    }
+
+    /**
+     * Opens the downloaded file
+     *
+     * @param id {string} File ID
+     */
+    public openFile(id: string): void {
+        const filePath = this.getFilePath(id);
+
+        const openResponse = fs.existsSync(`${filePath}`) && shell.openItem(`${filePath}`);
+        if (openResponse) {
+            return;
+        }
+
+        DownloadHandler.showDialog();
+    }
+
+    /**
+     * Opens the downloaded file in finder/explorer
+     *
+     * @param id {string} File ID
+     */
+    public showInFinder(id: string): void {
+        const filePath = this.getFilePath(id);
+
+        if (fs.existsSync(filePath)) {
+            shell.showItemInFolder(filePath);
+            return;
+        }
+
+        DownloadHandler.showDialog();
+    }
+
+    /**
+     * Clears download items
+     */
+    public clearDownloadItems(): void {
+        this.items = [];
+    }
+
+    /**
+     * Handle a successful download
+     * @param item Download item
+     */
+    public onDownloadSuccess(item: IDownloadManager): void {
+        let itemCount = 0;
+        for (const existingItem of this.items) {
+            if (item.fileName === existingItem.fileName) {
+                itemCount++;
+            }
+        }
+        item.count = itemCount;
+        item.fileDisplayName = DownloadHandler.getFileDisplayName(item.fileName, item);
+        this.items.push(item);
+        this.sendDownloadCompleted();
+    }
+
+    /**
+     * Handle a failed download
+     */
+    public onDownloadFailed(): void {
+        this.sendDownloadFailed();
+    }
+
+    /**
+     * Send download completed event to the renderer process
+     */
+    private sendDownloadCompleted(): void {
+        if (this.window && !this.window.isDestroyed()) {
+            logger.info(`download-handler: Download completed! Informing the client!`);
+            this.window.send('download-completed', this.items.map((item) => {
+                return {id: item._id, fileDisplayName: item.fileDisplayName};
```

Comment:
I think it would be better if we include the filesize

Reply:
Do we show file size today in the download manager?

Reply:
Yes